frontend/src/components/HuntDetails.tsx

```diff
@@ -0,0 +1,585 @@
+import { cn } from "@/lib/utils";
+import {
+  BsArrowLeft,
+  BsQrCode,
+  BsExclamationTriangle,
+  BsLink45Deg,
+} from "react-icons/bs";
+import { TbCalendarClock } from "react-icons/tb";
+import { GiSandsOfTime } from "react-icons/gi";
+import { HuddleRoom } from "./HuddleRoom";
+import { useReadContract, useActiveAccount, useSendTransaction } from "thirdweb/react";
+import { getContract, prepareContractCall } from "thirdweb";
+import { CONTRACT_ADDRESSES } from "../lib/utils";
+import { toast } from "sonner";
+import { client } from "../lib/client";
+import { baseSepolia, paseoAssetHub } from "../lib/chains";
+import QRCode from "react-qr-code";
+import QrScanner from "qr-scanner";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "./ui/tabs";
+import { generateInviteHash, encodeInviteToBase58, calculateInviteExpiry, decodeBase58Invite } from "../helpers/inviteUtils";
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { huntABI } from "../assets/hunt_abi.ts";
+
+// Type guard to ensure address is a valid hex string
+function isValidHexAddress(address: string): address is `0x${string}` {
+  return /^0x[0-9a-fA-F]{40}$/.test(address);
+}
+
+export function HuntDetails() {
+  const { huntId } = useParams();
+  const navigate = useNavigate();
+  // Team status state
+  const [showTeamInfo, setShowTeamInfo] = useState<boolean>(false);
+  
+  // Team management state
+  const [teamCode, setTeamCode] = useState<string>(""); // Default team code
+  const [activeTab, setActiveTab] = useState<"create" | "join" | "invite">("create");
+  const [hasCamera, setHasCamera] = useState(false);
+  const [scanResult, setScanResult] = useState<string | null>(null);
+  
+  // Multi-use invite state
+  const [inviteCode, setInviteCode] = useState<string | null>(null);
+  const [isGeneratingInvite, setIsGeneratingInvite] = useState(false);
+  const [showInviteWarning, setShowInviteWarning] = useState(false);
+  
+  // localStorage key for invite codes
+  const getInviteStorageKey = () => `invite_code_${huntId}_${userWallet}`;
+  
+  // Video reference for QR scanner
+  const videoRef = useRef<HTMLVideoElement>(null);
+  
+  // QR scanner instance
+  const [qrScanner, setQrScanner] = useState<QrScanner | null>(null);
+
+  const account = useActiveAccount();
+  const userWallet = account?.address;
+  const { mutate: sendTransaction } = useSendTransaction();
+
+  // Add this to get current network from localStorage
+  const currentNetwork = localStorage.getItem("current_network") || "assetHub";
+  const contractAddress =
+    CONTRACT_ADDRESSES[currentNetwork as keyof typeof CONTRACT_ADDRESSES] ??
+    "0x0000000000000000000000000000000000000000";
+
+  if (!isValidHexAddress(contractAddress)) {
+    toast.error("Invalid contract address format");
+    return null;
+  }
+
+  const contract = getContract({
+    client,
+    chain: currentNetwork === "base" ? baseSepolia : paseoAssetHub,
```

Comment:
There is a common helper for this in chains or lib/utils.ts
If we repeat this everywhere, it will make it hard to maintain